src/test/scala/SimpleKafkaConsumer.scala

```diff
@@ -0,0 +1,49 @@
+import java.util.Properties
+
+import com.fasterxml.jackson.databind.KeyDeserializer
+import org.apache.kafka.clients.consumer.{ConsumerRecords, KafkaConsumer}
+import org.apache.kafka.common.serialization.Deserializer
+import net.manub.embeddedkafka.Codecs.stringDeserializer
+import net.manub.embeddedkafka.ConsumerExtensions._
+
+class SimpleKafkaConsumer[K,V](consumerProps : Properties,
+                               topic : String,
+                               keyDeserializer: Deserializer[K],
+                               valueDeserializer: Deserializer[V],
+                               function : ConsumerRecords[K, V] => Unit,
+                               poll : Long = 2000) {
+
+  private var running = false
+
+  private val consumer = new KafkaConsumer[K, V](consumerProps, keyDeserializer, valueDeserializer)
+
+
+  private val thread = new Thread {
+    import scala.collection.JavaConverters._
+
+    override def run: Unit = {
+      consumer.subscribe(List(topic).asJava)
+      consumer.partitionsFor(topic)
+
+      while (running) {
```

Comment:
Its the way I've seen most of the other consumers written https://github.com/search?l=Scala&q=%22org.apache.kafka.clients.consumer.KafkaConsumer%22+%22while%22&type=Code&utf8=%E2%9C%93

I derived this from the "Automatic offset committing" section of https://kafka.apache.org/0101/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html

The poll is going to block up to  2000ms by default.